client/requestUtil.js

```diff
@@ -249,8 +271,30 @@ RequestUtil.prototype.list = function (category, startAt, maxRecords, nextContin
       WaitTimeSeconds: CONFIG.SQS_MESSAGES_LONGPOLL_TIMEOUT
     }
 
-    return s3Helper.listNotifications(this.sqs, notificationParams, category,
-      prefix)
+    // We will fetch from both old and new SQS queues until old one gets retired
+    if (this.oldSQSUrlByCat[category]) {
+      let oldNotificationParams = Object.assign({}, notificationParams)
+      oldNotificationParams.QueueUrl = `${this.oldSQSUrlByCat[category]}`
+
+      return s3Helper.listNotifications(
+        this.sqs, notificationParams, category, prefix).then((values) => {
+        if (this.shouldRetireOldSQSQueue(parseInt(values.createdTimeStamp))) {
+          return this.deleteSQSQueue(this.oldSQSUrlByCat[category]).then(() => {
```

Comment:
@AlexeyBarabash reminded me that when device id is duplicated, if there is an old Brave which doesn't contain the fix, its SQS queue will be unavailable until next relaunch because upgraded Brave deletes it after 24 hours

Reply:
I was confused and I thought `this.deleteSQSQueue` can delete queues belonging to other devices. And like @darkdh mentioned that only can happen if device id was duplicated. Which is a separate case.

Reply:
Non upgraded device will get reset anyway, so duplicate case won’t be an issue